### PR TITLE
Add XML Documentation to AzureSubscriptionExtension.cs

### PR DESCRIPTION
**Pull request by nisha-bhatia**

### CODE REVIEW

**Comment by nisha-bhatia** [comment]
ADO item: https://dev.azure.com/azure-mgmt-ex/DotNET%20Management%20SDK/_boards/board/t/DotNET%20Management%20SDK%20Team/Stories/?workitem=4897

File: azure-proto-compute/Extensions/AzureSubscriptionExtension.cs [hunk 1]
```diff
@@ -16,10 +16,10 @@ public static class AzureSubscriptionExtension
     {
         #region Virtual Machine List Operations
         /// <summary>
-        /// List vms at the given subscription context
+        /// List VMs at the given subscription.
```
Comment (m-nash) [inline azure-proto-compute/Extensions/AzureSubscriptionExtension.cs:19]: use standard text
https://github.com/m-nash/azure-proto-sdk/blob/master/XmlDocStandard.md#listresource--listresourceasync-on-subscriptionextensions

File: azure-proto-compute/Extensions/AzureSubscriptionExtension.cs [hunk 2]
```diff
@@ -16,10 +16,10 @@ public static class AzureSubscriptionExtension
...
         /// </summary>
-        /// <param name="subscription"></param>
-        /// <returns></returns>
+        /// <param name="subscription"> The id of the Azure subscription. </param>
```
Comment (m-nash) [inline azure-proto-compute/Extensions/AzureSubscriptionExtension.cs:21]: this isn't actually an id its the subscriptionoperation instance

File: azure-proto-compute/Extensions/AzureSubscriptionExtension.cs [hunk 3]
```diff
@@ -42,10 +42,10 @@ private static ComputeManagementClient GetComputeClient(SubscriptionOperations s
         }
 
         /// <summary>
-        /// 
+        /// List VMs at the given subscription.
```
Comment (m-nash) [inline azure-proto-compute/Extensions/AzureSubscriptionExtension.cs:45]: use standard text
https://github.com/m-nash/azure-proto-sdk/blob/master/XmlDocStandard.md#listresource--listresourceasync-on-subscriptionextensions

File: azure-proto-compute/Extensions/AzureSubscriptionExtension.cs [hunk 4]
```diff
@@ -56,13 +56,27 @@ public static AsyncPageable<VirtualMachine> ListVirtualMachinesAsync(this Subscr
                 s => new VirtualMachine(subscription.ClientOptions, new VirtualMachineData(s)));
         }
 
+        /// <summary>
+        /// List VMs by name at the given subscription.
```
Comment (m-nash) [inline azure-proto-compute/Extensions/AzureSubscriptionExtension.cs:60]: use standard text
https://github.com/m-nash/azure-proto-sdk/blob/master/XmlDocStandard.md#listresourcebyname--listresourcebynameasync-on-subscriptionextensions

File: azure-proto-compute/Extensions/AzureSubscriptionExtension.cs [hunk 5]
```diff
@@ -56,13 +56,27 @@ public static AsyncPageable<VirtualMachine> ListVirtualMachinesAsync(this Subscr
...
+        /// <returns> A response with the <see cref="Pageable{ArmResource}"/> operation for this resource. </returns>
         public static Pageable<ArmResource> ListVirtualMachinesByName(this SubscriptionOperations subscription, ArmSubstringFilter filter, int? top = null, CancellationToken cancellationToken = default)
         {
             ArmFilterCollection filters = new ArmFilterCollection(VirtualMachineOperations.ResourceType);
             filters.SubstringFilter = filter;
             return ResourceListOperations.ListAtContext(subscription.ClientOptions, subscription.Id, filters, top, cancellationToken);
         }
 
+        /// <summary>
+        /// List VMs by name at the given subscription.
```
Comment (m-nash) [inline azure-proto-compute/Extensions/AzureSubscriptionExtension.cs:74]: https://github.com/m-nash/azure-proto-sdk/blob/master/XmlDocStandard.md#listresourcebyname--listresourcebynameasync-on-subscriptionextensions

File: azure-proto-compute/Extensions/AzureSubscriptionExtension.cs [hunk 6]
```diff
@@ -72,6 +86,11 @@ public static AsyncPageable<ArmResource> ListVirtualMachinesByNameAsync(this Sub
         #endregion
 
         #region AvailabilitySet List Operations
+        /// <summary>
+        /// List AvailabilitySets at the given subscription.
```
Comment (m-nash) [inline azure-proto-compute/Extensions/AzureSubscriptionExtension.cs:90]: use standard text
https://github.com/m-nash/azure-proto-sdk/blob/master/XmlDocStandard.md#listresource--listresourceasync-on-subscriptionextensions

File: azure-proto-compute/Extensions/AzureSubscriptionExtension.cs [hunk 7]
```diff
@@ -82,6 +101,11 @@ public static Pageable<AvailabilitySet> ListAvailabilitySets(this SubscriptionOp
                 s => new AvailabilitySet(subscription.ClientOptions, new AvailabilitySetData(s)));
         }
 
+        /// <summary>
+        /// List AvailabilitySets at the given subscription.
```
Comment (m-nash) [inline azure-proto-compute/Extensions/AzureSubscriptionExtension.cs:105]: use standard text
https://github.com/m-nash/azure-proto-sdk/blob/master/XmlDocStandard.md#listresource--listresourceasync-on-subscriptionextensions

**Review by m-nash — CHANGES_REQUESTED** [review]

**Review by m-nash — APPROVED** [review]

**Pull request merged**